transformations/transformations/pool_allocator.py

```diff
@@ -579,7 +580,7 @@ def create_pool_allocator(self, routine, stack_size):
                 if pragma.content.lower().startswith('parallel') and 'gang' in pragma.content.lower():
                     parameters = get_pragma_parameters(pragma, starts_with='parallel', only_loki_pragmas=False)
                     if 'private' in [p.lower() for p in parameters]:
-                        content = pragma.content.lower().replace('private(', f'private({stack_var.name}, ')
+                        content = re.sub(r'\bprivate\(', f'private({stack_var.name}, ', pragma.content.lower())
```

Comment:
[no action] Is this just to catch optional whitespace? Just for my own understanding, why is this an issue?

Reply:
If the pragma had both a `private` and a `firstprivate` clause, then the previous implementation would put `YLSTACK` in both.

Reply:
Ahhhhh! Makes sense, thanks.